### PR TITLE
[CARF-412] Implement Update subscription endpoint

### PR DESCRIPTION
**Pull request by ZDevelop94**

### CODE REVIEW

File: app/uk/gov/hmrc/carfregistration/controllers/SubscriptionController.scala [hunk 1]
```diff
@@ -55,6 +55,29 @@ class SubscriptionController @Inject() (
...
+      .validate[SubscriptionRequest]
+      .fold(
+        invalid = _ =>
+          Future.successful(
+            BadRequest("Update Subscription Request's Json is invalid")
+          ),
+        valid = subscription =>
+          subscriptionConnector.updateSubscription(subscription).value.map {
+            case Right(httpResponse)                =>
+              Status(httpResponse.status)(httpResponse.body)
```
Comment (simondrugan16) [inline app/uk/gov/hmrc/carfregistration/controllers/SubscriptionController.scala]: Is Status(httpResponse.status) better than just doing Ok()? To me, it makes more sense to be to be explicit on what our BE sends to our FE (since it will always be 200 anyway due to the connector, and if that changed, code changes would be isolated to the BE only)

File: app/uk/gov/hmrc/carfregistration/connectors/SubscriptionConnector.scala [hunk 2]
```diff
@@ -127,15 +155,17 @@ class SubscriptionConnector @Inject() (
     }
   }
 
-  private def logDownStreamError(status: Int, body: String): Unit = {
+  private def logDownStreamError(status: Int, body: String): Option[ErrorDetail] = {
```
Comment (simondrugan16) [inline app/uk/gov/hmrc/carfregistration/connectors/SubscriptionConnector.scala:158]: We should've just done it like this originally tbh :D code tries to parse all errors, if it can't, it doesn't matter

File: it/test/connectors/SubscriptionConnectorISpec.scala [hunk 3]
```diff
@@ -267,6 +270,51 @@ class SubscriptionConnectorISpec
...
+              .withBody(testSubscriptionResponseJson)
+          )
+      )
+
+      val result = connector.updateSubscription(testSubscriptionRequest).value.futureValue
+
+      result.map(_.body) mustBe Right(testSubscriptionResponseJson)
+    }
+
+    "return Left InternalServerError if BAD_REQUEST status response is returned from backend" in {
```
Comment (simondrugan16) [inline it/test/connectors/SubscriptionConnectorISpec.scala]: One more test where Option[ErrorDetail] is None perhaps?

**Review by simondrugan16 — APPROVED** [review]
